src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -4097,6 +4252,11 @@ public List<TElement> ToList()
             return list;
         }
 
+        public int Count
+        {
+            get { return source.Count(); }
+        }
```

Comment:
As in the other, it's always cheaper than what is currently happening, if not necessarily cheap enough to base some optimisations on top of. It's always a win for `Count()` to hit it and makes other cases that are a bigger win simpler, and it's not currently at risk of being used elsewhere.